src/streamvbyte.c

```diff
@@ -311,12 +387,23 @@ const uint8_t *svb_decode_avx_simple(uint32_t *out,
 size_t streamvbyte_decode(const uint8_t *in, uint32_t *out, uint32_t count) {
   if (count == 0)
     return 0;
+
   const uint8_t *keyPtr = in;               // full list of keys is next
   uint32_t keyLen = ((count + 3) / 4);      // 2-bits per key (rounded up)
   const uint8_t *dataPtr = keyPtr + keyLen; // data starts at end of keys
+
 #ifdef __AVX__
-  return svb_decode_avx_simple(out, keyPtr, dataPtr, count) - in;
-#else
-  return svb_decode_scalar(out, keyPtr, dataPtr, count) - in;
+  dataPtr = svb_decode_avx_simple(out, keyPtr, dataPtr, count);
```

Comment:
This is also an attempt to inline the vector decoding and scalar tail-processing.  The AVX and NEON versions have different vector lengths, so we figure the lengths in each version.